.github/workflows/luarocks.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: Push to Luarocks
+
+on:
+  push:
+    tags:
+      - '*'
+  release:
+    types:
+      - created
+    tags:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  luarocks-upload:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Required to count the commits
+      - name: Get Version
+        run: echo "LUAROCKS_VERSION=$(git describe --abbrev=0 --tags)" >> $GITHUB_ENV
+      - name: LuaRocks Upload
+        uses: nvim-neorocks/luarocks-tag-release@v5
+        env:
+          LUAROCKS_API_KEY: ${{ secrets.LUAROCKS_API_KEY }}
+        with:
+          version: ${{ env.LUAROCKS_VERSION }}
+          dependencies: |
+            nvim-web-devicons
+            gitsigns.nvim
```

Comment:
Is there a way to express optional dependencies in the neorocks CI job? We have integrations with these plugins but users _need_ them

If not, don't worry about it 😁 we can leave it as-is

Reply:
No, luarocks doesn't have a concept of optional dependencies.
If the user experience or UI is improved, I would suggest keeping them, because they do not impact startup time in any noticeable way.
See also: https://github.com/nvim-neorocks/rocks.nvim?tab=readme-ov-file#a-note-on-loading-rocks

Reply:
Got it— thanks for the link!